### PR TITLE
Add Views and Templates from View-template to master

### PR DESCRIPTION
**Pull request by bragadeeshwaran06**
This pull request introduces the following changes from the View-template branch to the master branch:
1.New Views Added
2.New Templates Added
3.New Urls 
4.New Forms

### CODE REVIEW

File: classifides/ads/views.py [hunk 1]
```diff
@@ -1,2 +1,136 @@
...
+
+    return render(request, 'ad_detail.html', {
+        'ad': ad,
+        'ad_images': ad_images,
+        'categories': categories,
+        'liked_by_user': liked_by_user, 
+    })
+
+@login_required
+def like_ad(request, ad_id):
```
Comment (shantanutomarr) [inline classifides/ads/views.py:55]: This should not accept GET request. Only post should be accepted. Any request that is adding changes to the database should be a POST request.
GET is only for read-only operations.

File: classifides/ads/views.py [hunk 2]
```diff
@@ -1,2 +1,136 @@
...
+@login_required
+def delete_ad(request, pk):
+    ad = get_object_or_404(Ad, id=pk)
+    
+    if request.method == 'POST':
+        ad.delete()
+        messages.success(request, 'Ad deleted successfully.')
+        return redirect('my_ads')  
+
+    return render(request, 'ad_confirm_delete.html', {'ad': ad})
```
Comment (shantanutomarr) [inline classifides/ads/views.py]: Why two views are added for deleting ad ? I think this is by mistake. Please remove one.

File: classifides/ads/views.py [hunk 3]
```diff
@@ -1,2 +1,136 @@
...
+
+            return redirect('ad_detail', ad_id=ad.id)
+    else:
+        form = AdForm(instance=ad)
+        formset = AdImageFormSet(instance=ad)
+
+    return render(request, 'ad_edit.html', {
+        'form': form,
+        'formset': formset,
+    })
```
Comment (shantanutomarr) [inline classifides/ads/views.py]: In case of Update view, I should not be allowed to edit the ads created by other users.

If I try to edit other user's ad, you can raise 404 error.

**Review by shantanutomarr — CHANGES_REQUESTED** [review]

File: classifides/ads/views.py (hunk 1 above)
Comment (bragadeeshwaran06) [reply classifides/ads/views.py:55]: Ok

File: classifides/ads/views.py (hunk 2 above)
Comment (bragadeeshwaran06) [reply classifides/ads/views.py]: Sure

File: classifides/ads/views.py (hunk 3 above)
Comment (bragadeeshwaran06) [reply classifides/ads/views.py]: Ok

File: classifides/ads/context_processors.py [hunk 4]
```diff
@@ -0,0 +1,6 @@
+from .models import Category
+
+
+def category_processor(request):
```
Comment (shantanutomarr) [inline classifides/ads/context_processors.py:4]: Where is this used ? Please remove this if not used.

File: classifides/ads/views.py [hunk 5]
```diff
@@ -1,2 +1,136 @@
...
+        'ad': ad,
+        'ad_images': ad_images,
+        'categories': categories,
+        'liked_by_user': liked_by_user,
+        'receiver': receiver, 
+    })
+
+@login_required
+def like_ad(request, ad_id):
+    if request.method == 'POST':
```
Comment (shantanutomarr) [inline classifides/ads/views.py]: Instead of checking with IF block, you can use `require_POST` decorator on the view.
https://docs.djangoproject.com/en/5.1/topics/http/decorators/#django.views.decorators.http.require_POST

File: classifides/ads/views.py [hunk 6]
```diff
@@ -1,2 +1,136 @@
...
+        'form': form,
+        'formset': formset,
+    })
+
+
+@login_required
+def delete_ad(request, pk):
+    ad = get_object_or_404(Ad, id=pk)
+    
+    if request.method == 'POST':
```
Comment (shantanutomarr) [inline classifides/ads/views.py]: - use require_POST decorator
- And here also I should not be able to delete ad posted by other users.

**Review by shantanutomarr — CHANGES_REQUESTED** [review]

File: classifides/ads/views.py (hunk 6 above)
Comment (bragadeeshwaran06) [reply classifides/ads/views.py]: ok

File: classifides/ads/context_processors.py (hunk 4 above)
Comment (bragadeeshwaran06) [reply classifides/ads/context_processors.py:4]: It is used in the nav bar of all the templates

File: classifides/ads/views.py (hunk 6 above)
Comment (bragadeeshwaran06) [reply classifides/ads/views.py:130]: ok

File: classifides/ads/views.py (hunk 5 above)
Comment (bragadeeshwaran06) [reply classifides/ads/views.py]: ok

File: classifides/classifides/settings.py [hunk 7]
```diff
@@ -30,17 +31,25 @@
...
     'django.contrib.messages',
     'django.contrib.staticfiles',
-    'ads',
+    'ads.apps.AdsConfig',
     'taggit',
+    'accounts.apps.AccountsConfig',
+    'django.contrib.humanize',
+    'widget_tweaks',
+    'channels',
+    'chat.apps.ChatConfig',
```
Comment (shantanutomarr) [inline classifides/classifides/settings.py]: Remove these from this PR. This is not related to this PR.

File: classifides/classifides/settings.py [hunk 8]
```diff
@@ -101,6 +111,14 @@
...
 ]
 
+CHANNEL_LAYERS = {
+    'default': {
+        'BACKEND': 'channels_redis.core.RedisChannelLayer',
+        'CONFIG': {
+            "hosts": [('127.0.0.1', 6379)],
+        },
+    },
+}
```
Comment (shantanutomarr) [inline classifides/classifides/settings.py]: This is not related to this PR.

File: classifides/classifides/settings.py [hunk 9]
```diff
@@ -82,7 +92,7 @@
     }
 }
 
-
+ASGI_APPLICATION = 'classifides.asgi.application'
```
Comment (shantanutomarr) [inline classifides/classifides/settings.py]: This is not related to this PR.

**Review by shantanutomarr — CHANGES_REQUESTED** [review]

**Review by shantanutomarr — APPROVED** [review]
